### PR TITLE
[Trello 166] Dating Cards

### PR DESCRIPTION
**Pull request by azimgd**

### CODE REVIEW

File: src/components/AuthSigninEmail/Form.js [hunk 1]
```diff
@@ -15,13 +15,13 @@ import testIDs from './test-ids'
 const formSchema = Yup.object().shape({
   username: Yup.string()
     .min(3)
-    .max(50)
+    .max(80)
```
Comment (omelniz) [inline src/components/AuthSigninEmail/Form.js:18]: As I remember we have 50 max length validation on the backend. I guess you changed it to use generated staging users, maybe it is better to make email shorter for that users

File: src/components/Dating/About.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    fontWeight: '400',
+    color: theme.colors.border,
+  },
+})
+
+About.propTypes = {
+  t: PropTypes.any,
+  theme: PropTypes.any,
+  user: PropTypes.any,
+  flingPosition: PropTypes.any,
```
Comment (omelniz) [inline src/components/Dating/About.js:74]: oneOf(['DOWN', 'UP'])

File: src/components/Dating/Placeholder.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  theme,
+  datingMatchedUsersRequest,
+}) => {
+  const styling = styles(theme)
+  const navigation = useNavigation()
+
+  return (
+    <View style={styling.root}>
+      <View style={styling.placeholder}>
+        <Text style={styling.text}>We cannot find potential matches or you have viewed them all based on your search criteria.</Text>
```
Comment (omelniz) [inline src/components/Dating/Placeholder.js:26]: t('

File: src/components/Dating/index.js [hunk 4]
```diff
@@ -1,90 +1,125 @@
...
 })
 
 Dating.propTypes = {
   theme: PropTypes.any,
-  t: PropTypes.any,
   user: PropTypes.any,
-  usersSetUserDatingStatusRequest: PropTypes.any,
-  isDatingAvailable: PropTypes.any,
-  isDatingAvailableDebug: PropTypes.any,
+  datingMatchedUsersRequest: PropTypes.any,
```
Comment (omelniz) [inline src/components/Dating/index.js:118]: .func

File: src/components/DatingMatch/Form.js [hunk 5]
```diff
@@ -57,11 +57,10 @@ const DatingMatchForm = ({
    * [matchLocationRadius] Match radius
    */
   const locationItems = [
-    { label: '5 km', value: 5 },
-    { label: '10 km', value: 10 },
-    { label: '30 km', value: 30 },
-    { label: '50 km', value: 50 },
-    { label: '100 km', value: 100 },
+    { label: '15 mi', value: 10 },
```
Comment (omelniz) [inline src/components/DatingMatch/Form.js:60]: 10 !== 15 mi

File: src/components/DatingMatch/index.service.js [hunk 6]
```diff
@@ -31,21 +31,20 @@ const DatingMatchService = ({ children }) => {
...
   }
 
   const formSubmitLoading = usersEditProfile.status === 'loading'
   const formSubmitDisabled = usersEditProfile.status === 'loading'
   const formErrorMessage = usersEditProfile.error.text
 
   const formInitialValues = {
-    matchAgeRangeMin: user.matchAgeRange.min || 18,
-    matchAgeRangeMax: user.matchAgeRange.max || 21,
+    matchAgeRangeMin: (user.matchAgeRange || {}).min || 18,
```
Comment (omelniz) [inline src/components/DatingMatch/index.service.js:46]: Could we https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining here?

File: src/components/DatingProfile/index.js [hunk 7]
```diff
@@ -4,45 +4,60 @@ import {
...
   },
 })
 
 DatingProfile.propTypes = {
-  theme: PropTypes.any,
   t: PropTypes.any,
+  theme: PropTypes.any,
+  user: PropTypes.any,
+  usersSetUserDatingStatus: PropTypes.any,
+  usersSetUserDatingStatusRequest: PropTypes.any,
```
Comment (omelniz) [inline src/components/DatingProfile/index.js:60]: .func

File: src/components/Formik/MapField.js [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+  const getCurrentPosition = () => new Promise((resolve, reject) => {
+    Geolocation.getCurrentPosition(
+      compose(resolve, normalizeCoordinates, prop('coords')),
+      reject,
+      { timeout: 2000, maximumAge: 1000 },
+    )
+  })
+
+  const coordinates = useAsync(async () => await getCurrentPosition(), [])
```
Comment (omelniz) [inline src/components/Formik/MapField.js:43]: nice

File: src/navigation/Profile/index.js [hunk 9]
```diff
@@ -93,7 +96,21 @@ const ProfileNavigator = () => {
...
+      />
+      <Stack.Screen
+        name="DatingMatch"
+        component={DatingMatchScreen}
+        {...stackScreenPageProps({ options: { title: 'Dating Preferences' } })}
+      />
+      <Stack.Screen
+        name="DatingProfile"
+        component={DatingProfileScreen}
+        {...stackScreenPageProps({ options: { title: 'Dating Profile' } })}
```
Comment (omelniz) [inline src/navigation/Profile/index.js:112]: someday we will need to add t('') here

File: src/store/ducks/dating/constants.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
+export const DATING_MATCHED_USERS_REQUEST = 'DATING_MATCHED_USERS_REQUEST'
+export const DATING_MATCHED_USERS_SUCCESS = 'DATING_MATCHED_USERS_SUCCESS'
+export const DATING_MATCHED_USERS_FAILURE = 'DATING_MATCHED_USERS_FAILURE'
+export const DATING_MATCHED_USERS_LOADING = 'DATING_MATCHED_USERS_LOADING'
+export const DATING_MATCHED_USERS_IDLE = 'DATING_MATCHED_USERS_IDLE'
+
+export const DATING_CONFIRMED_USERS_REQUEST = 'DATING_CONFIRMED_USERS_REQUEST'
+export const DATING_CONFIRMED_USERS_SUCCESS = 'DATING_CONFIRMED_USERS_SUCCESS'
+export const DATING_CONFIRMED_USERS_FAILURE = 'DATING_CONFIRMED_USERS_FAILURE'
+export const DATING_CONFIRMED_USERS_LOADING = 'DATING_CONFIRMED_USERS_LOADING'
```
Comment (omelniz) [inline src/store/ducks/dating/constants.js:10]: I've found a place where we use the variable

File: src/store/ducks/dating/saga/datingConfirmedUsers.js [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+  const meta = metaSelector(api)
+  const payload = req.payload
+
+  const normalized = normalizer.normalizeUsersGet(data)
+  yield put(entitiesActions.entitiesAlbumsMerge({ data: normalized.entities.albums || {} }))
+  yield put(entitiesActions.entitiesPostsMerge({ data: normalized.entities.posts || {} }))
+  yield put(entitiesActions.entitiesUsersMerge({ data: normalized.entities.users || {} }))
+  yield put(entitiesActions.entitiesCommentsMerge({ data: normalized.entities.comments || {} }))
+  yield put(entitiesActions.entitiesImagesMerge({ data: normalized.entities.images || {} }))
+
```
Comment (omelniz) [inline src/store/ducks/dating/saga/datingConfirmedUsers.js:29]: we can move that logic to entities/saga.js
`
function* entitiesMerge(data) {
  yield put(entitiesActions.entitiesAlbumsMerge({ data: normalized.entities.albums || {} }))
  yield put(entitiesActions.entitiesPostsMerge({ data: normalized.entities.posts || {} }))
  yield put(entitiesActions.entitiesUsersMerge({ data: normalized.entities.users || {} }))
  yield put(entitiesActions.entitiesCommentsMerge({ data: normalized.entities.comments || {} }))
  yield put(entitiesActions.entitiesImagesMerge({ data: normalized.entities.images || {} }))
}`

File: src/store/ducks/dating/selectors.js [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+const createDeepEqualSelector = createSelectorCreator(
+  defaultMemoize,
+  equals,
+)
+
+/**
+ *
+ */
+const datingMatchedUsers = () => path(['dating', 'datingMatchedUsers'])
```
Comment (omelniz) [inline src/store/ducks/dating/selectors.js:16]: no need to () =>

the path is curried and returns a fuction

File: src/components/DatingMatch/index.service.js (hunk 6 above)
Comment (omelniz) [reply src/components/DatingMatch/index.service.js:46]: pathOr

File: src/components/AuthSigninEmail/Form.js (hunk 1 above)
Comment (azimgd) [reply src/components/AuthSigninEmail/Form.js:18]: https://docs.aws.amazon.com/cognito/latest/developerguide/limits.html
```
Maximum characters in email subject | 140
```

We can actually increase it further

File: src/store/ducks/dating/constants.js (hunk 10 above)
Comment (azimgd) [reply src/store/ducks/dating/constants.js:10]: This was required for the first scope to show (bi-directional) matched users, but than we integrated chats. This logic will live under dating screen soon.

File: src/store/ducks/dating/saga/datingConfirmedUsers.js (hunk 11 above)
Comment (azimgd) [reply src/store/ducks/dating/saga/datingConfirmedUsers.js:29]: separate PR should be implemented for this

**Comment by azimgd** [comment]
@omelniz thanks, pushed

**Pull request merged**